### ISSUE TITLE
in-progress dehydrated meat has a long shelf life

### ISSUE THREAD

**Issue by factoryman942** [CleverRaven/Cataclysm-DDA#49718]
# Describe the bug

Beginning the craft for dehydrated meat immediately changes its displayed shelf life to 3 seasons.

# Steps To Reproduce

1.  Prepare a craft of dehydrated meat.
2.  Do not complete the craft.

# Expected behavior

The meat should keep its original rot time of 1 day until the craft is complete. Since smokerack/dehydrator charges are used over time during the craft, leaving it at 0% also means you can "loan" charcoal - start the craft, and you don't have to pay the charcoal cost until you want to use the meat.

# Screenshots

![image](https://user-images.githubusercontent.com/20996272/124808807-4fdd8b00-df57-11eb-8be6-4d79db528bd1.png)

# Versions and configuration

- OS: Windows
    - OS Version: 10.0 1909
- Game Version: bc28921 [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Bionic Professions [package_bionic_professions],
    No Fungal Monsters [No_Fungi],
    Stats Through Skills [StatsThroughSkills],
    Alternative Map Key [alt_map_key]
]

# Additional context

**Comment by robotic-gamer** [comment]
this seems to apply to everything (ex in progress tainted tallow doesn't rot). the closest thing to a true solution would be for the time to rot be between the raw materials and finished item. with the weight being the percentage done crafted. finished items that don't rot could have some arbitrary long shelf life (like 1000 days) as a placeholder.

**Comment by actual-nh** [comment]
Just be sure that the time doesn't become less than needed for the dehydration...

**Comment by robotic-gamer** [comment]
it shouldn't really occur in practice, 
tainted tallow adds 5 days with 1%, 
dehydrated meat adds 1.35 days with 1%,
meat jerky adds adds 4.5 hours with 1 %
assuming the material is fresh, half to gone off would add about half the time. still the shelf life should outstrip time spent cooking very quickly

**Comment by scorpion451** [comment]
Meat preservation is already heavily skewed in the wrong direction regarding freshness- apparently no one here has heard of meat hanging, the process of letting a carcass cure for a few days (or weeks, in modern refrigerated processes) before butchering. This would just exacerbate the issues already present.

**Comment by MomoiroShinobinomono** [comment]
> Meat preservation is already heavily skewed in the wrong direction regarding freshness- apparently no one here has heard of meat hanging, the process of letting a carcass cure for a few days (or weeks, in modern refrigerated processes) before butchering. This would just exacerbate the issues already present.

I agree with you, corpses that rot to the bone in mere hours is stupid but thing is, how in the world the game is supposed to check if the conditions for dry or wet aging are favourable?
It's not just a matter of ambient temperature or meat being hung inside or outside.

**Comment by Aerotinge** [comment]
Boiling water, smoke or dehydrate meats, these time consuming crafting indeed need split into 2 or more steps. 
And character should be able to leave boring waiting time there and return when it was done.
Currently, I think this exploit is OK. If you ever try batch work of cooking offal/meat, some of them even rot in the middle of crafting. I have to exploit these dehydrated meat and use them as ingredient later, really time wasting.
